pkg/controller/pxc/users.go

```diff
@@ -326,3 +327,39 @@ func sysUsersSecretDataChanged(newHash string, usersSecret *corev1.Secret) (bool
 func sha256Hash(data []byte) string {
 	return fmt.Sprintf("%x", sha256.Sum256(data))
 }
+
+func (r *ReconcilePerconaXtraDBCluster) reconcileSyncPXCUsersWithProxySQL(cr *api.PerconaXtraDBCluster) {
+	if cr.Status.Status != api.AppStateReady {
+		return
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	stopCh := make(chan error, 1)
+
+	go func() {
+		r.syncUsersChan <- 1
```

Comment:
I propose to use an atomic CAS operation instead of the channel for that reason.